File: app/src/main/java/com/slapshotapps/dragonshockey/models/SeasonSchedule.java
package com.slapshotapps.dragonshockey.models;

import android.support.annotation.Keep;
import java.util.ArrayList;
import java.util.List;

/**
 * A class that will hold the schedule for the season
 */
@Keep
public class SeasonSchedule {
    public List<Game> games;

    public SeasonSchedule() {
        games = new ArrayList<>();
    }

    public void addGame(Game game) {
        games.add(game);
    }

    public List<Game> getAllGames() {
        return games;
    }

    public int numberOfGames() {
        return games.size();
    }

    public Game getGame(int gameID) {
        for (Game game : games) {
            if (game.gameID == gameID) {
                return game;
            }
        }

        return null;
    }
}

File: app/src/main/java/com/slapshotapps/dragonshockey/models/SeasonSchedule.kt
package com.slapshotapps.dragonshockey.models

import androidx.annotation.Keep
import java.util.*

/**
 * A class that will hold the schedule for the season
 */
@Keep
class SeasonSchedule {
    var games: MutableList<Game> = ArrayList()

    val allGames: List<Game>
        get() = games

    fun addGame(game: Game) {
        games.add(game)
    }

    fun numberOfGames(): Int {
        return games.size
    }

    fun getGame(gameID: Int): Game? {
        return games.first {
            gameID == it.gameID
        }
    }
}
